src/java.base/share/classes/javax/crypto/spec/HPKEParameterSpec.java

```diff
@@ -0,0 +1,480 @@
+ *      does not support authentication.
+ * </ul>
+ * Example:
+ * {@snippet lang = java:
+ * // Key pair generation
+ * KeyPairGenerator g = KeyPairGenerator.getInstance("X25519");
+ * KeyPair kp = g.generateKeyPair();
+ *
+ * // The HPKE sender side is initialized with the recipient's public key
+ * Cipher sender = Cipher.getInstance("HPKE");
+ * HPKEParameterSpec ps = HPKEParameterSpec.of()
+ *         .info("this_info".getBytes(StandardCharsets.UTF_8));
+ * sender.init(Cipher.ENCRYPT_MODE, kp.getPublic(), ps);
+ *
+ * // Retrieve the key encapsulation message (the KEM output) from the sender
+ * byte[] kemEncap = sender.getIV();
+ *
+ * // The HPKE recipient side is initialized with its own private key
+ * // and the key encapsulation message from the sender
+ * Cipher recipient = Cipher.getInstance("HPKE");
+ * HPKEParameterSpec pr = HPKEParameterSpec.of()
+ *         .info("this_info".getBytes(StandardCharsets.UTF_8))
+ *         .encapsulation(kemEncap);
+ * recipient.init(Cipher.DECRYPT_MODE, kp.getPrivate(), pr);
+ *
+ * // Secure communication between the 2 sides
+ * byte[] msg = "Hello World".getBytes(StandardCharsets.UTF_8);
+ * byte[] ct = sender.doFinal(msg);
+ * byte[] pt = recipient.doFinal(ct);
+ *
+ * assert Arrays.equals(msg, pt);
+ * }
+ *
+ * @implNote
+ * In the HPKE implementation in the SunJCE provider included in this JDK
+ * implementation, {@code HPKEParameterSpec.of()} chooses the following
+ * KEM, KDF, and AEAD algorithms depending on the provided key type:
+ * <table class="striped">
+ * <caption style="display:none">Default Algorithm Identifiers</caption>
+ * <thead>
+ * <tr><th scope="col">key type
+ *     <th scope="col">{@code kem_id}
+ *     <th scope="col">{@code kdf_id}
+ *     <th scope="col">{@code aead_id}
+ * </thead>
+ * <tbody>
+ * <tr><td>EC (secp256r1)
+ *     <td>0x10<br>DHKEM(P-256, HKDF-SHA256)
+ *     <td>0x1<br>HKDF-SHA256
+ *     <td rowspan="5">0x2<br>AES-256-GCM
+ * <tr><td>EC (secp384r1)
+ *     <td>0x11<br>DHKEM(P-384, HKDF-SHA384)
+ *     <td>0x2<br>HKDF-SHA384
+ * <tr><td>EC (secp521r1)
+ *     <td>0x12<br>DHKEM(P-521, HKDF-SHA512)
+ *     <td>0x3<br>HKDF-SHA512
+ * <tr><td>XDH (X25519)
+ *     <td>0x20<br>DHKEM(X25519, HKDF-SHA256)
+ *     <td>0x1<br>HKDF-SHA256
+ * <tr><td>XDH (X448)
+ *     <td>0x21<br>DHKEM(X448, HKDF-SHA512)
+ *     <td>0x3<br>HKDF-SHA512
+ * </tbody>
+ * </table>
+ * No other keys are supported.
+ *
+ * @spec https://www.rfc-editor.org/info/rfc9180
+ *      RFC 9180: Hybrid Public Key Encryption
+ * @spec security/standard-names.html
+ *      Java Security Standard Algorithm Names
+ * @since 25
+ */
+public final class HPKEParameterSpec implements AlgorithmParameterSpec {
+
+    /**
+     * KEM algorithm identifier for DHKEM(P-256, HKDF-SHA256) as defined in
+     * <a href="https://www.rfc-editor.org/rfc/rfc9180.html#name-key-encapsulation-mechanism">Section 7.1 of RFC 9180</a>.
+     */
+    public static final int KEM_DHKEM_P_256_HKDF_SHA256 = 0x10;
```

Comment:
Do we really need `KEM_` at the beginning? Seems redundant since `DHKEM` has "KEM" in it. Same comment for the KDF constants.

Reply:
I was hoping that IDE auto-completion would eliminate the need to type the full name. Adding `KEM` at the beginning could amend the type-unsafety of current algorithm identifiers -- it somehow prevents putting a KDF id as the first argument of an `of` call accidentally. Maybe that is just my wishful thinking?

Also, there will be KEM names like "X_WING" in the future.

I've also thought about defining `enum`s for the three types of algorithm identifiers to finally solve the type-unsafety issue. However, I hesitate to add them since it's such a think layer and users who need it can easily create their own.

Reply:
Makes sense. Cut and paste is easy enough as well.